includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,488 @@
+	}
+
+	/**
+	 * WooCommerce Init
+	 */
+	public function woocommerce_init() {
+		include_once( dirname( __FILE__ ) . '/class-wc-background-swedbank-pay-queue.php' );
+		self::$background_process = new WC_Background_Swedbank_Pay_Queue();
+	}
+
+	/**
+	 * WooCommerce Loaded: load classes
+	 */
+	public function woocommerce_hook_loaded() {
+		// Includes
+	}
+
+	/**
+	 * Register payment gateway
+	 *
+	 * @param string $class_name
+	 */
+	public static function register_gateway( $class_name ) {
+		global $px_gateways;
+
+		if ( ! $px_gateways ) {
+			$px_gateways = array();
+		}
+
+		if ( ! isset( $px_gateways[ $class_name ] ) ) {
+			// Initialize instance
+			$gateway = new $class_name;
+
+			if ( $gateway ) {
+				$px_gateways[] = $class_name;
+
+				// Register gateway instance
+				add_filter(
+					'woocommerce_payment_gateways',
+					function ( $methods ) use ( $gateway ) {
+						$methods[] = $gateway;
+
+						return $methods;
+					}
+				);
+			}
+		}
+	}
+
+	/**
+	 * Allow processing/completed statuses for capture
+	 *
+	 * @param array $statuses
+	 * @param WC_Order $order
+	 *
+	 * @return array
+	 */
+	public function add_valid_order_statuses( $statuses, $order ) {
+		$payment_method = $order->get_payment_method();
+		if ( in_array( $payment_method, self::PAYMENT_METHODS, true ) ) {
+			$statuses = array_merge(
+				$statuses,
+				array(
+					'processing',
+					'completed',
+				)
+			);
+		}
+
+		return $statuses;
+	}
+
+	/**
+	 * Order Status Change: Capture/Cancel
+	 *
+	 * @param $order_id
+	 * @param $from
+	 * @param $to
+	 * @param $order
+	 */
+	public static function order_status_changed( $order_id, $from, $to, $order ) {
+		// We are need "on-hold" only
+		if ( 'on-hold' !== $from ) {
+			return;
+		}
+
+		// Disable status change hook
+		remove_action(
+			'woocommerce_order_status_changed',
+			'\SwedbankPay\Payments\WooCommerce\WC_Swedbank_Plugin::order_status_changed',
+			10
+		);
+
+		$payment_method = $order->get_payment_method();
+		if ( ! in_array( $payment_method, self::PAYMENT_METHODS, true ) ) {
+			return;
+		}
+
+		// Get Payment Gateway
+		$gateways = WC()->payment_gateways()->get_available_payment_gateways();
+
+		/** @var \WC_Gateway_Swedbank_Pay_Cc $gateway */
+		$gateway = $gateways[ $payment_method ];
+
+		switch ( $to ) {
+			case 'cancelled':
+				// Cancel payment
+				try {
+					$gateway->cancel_payment( $order );
+				} catch ( Exception $e ) {
+					$message = $e->getMessage();
+					WC_Admin_Meta_Boxes::add_error( $message );
+
+					// Rollback
+					$order->update_status(
+						$from,
+						/* translators: 1: note */ sprintf( __( 'Order status rollback. %1$s', 'swedbank-pay-woocommerce-payments' ), $message )
+					);
+				}
+				break;
+			case 'processing':
+			case 'completed':
+				// Capture payment
+				try {
+					// Capture
+					$gateway->capture_payment( $order );
+				} catch ( Exception $e ) {
+					$message = $e->getMessage();
+					WC_Admin_Meta_Boxes::add_error( $message );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class '\WC_Admin_Meta_Boxes' in method 'order_status_changed'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-checkout/pullRequest?prid=5510424)